src/OrleansProviders/Storage/MemoryStorage.cs

```diff
@@ -108,43 +108,34 @@ public virtual Task Close()
             
             string id = HierarchicalKeyStore.MakeStoreKey(keys);
             IMemoryStorageGrain storageGrain = GetStorageGrain(id);
-            Tuple<IDictionary<string, object>, string> result = await storageGrain.ReadStateAsync(STATE_STORE_NAME, id);
-            if (result != null && result.Item1 != null)
+            var state = await storageGrain.ReadStateAsync(STATE_STORE_NAME, id);
+            if (state != null)
```

Comment:
I don't think you want to change that semantics (not checking `result.Item1 != null`). It may or may not be correct, but that is what azure table storage is doing now, and we want to be identical on the semantics (even if they are wrong now). We should separably revisit the semantic change, for both.